pkg/generators/rules/namesMatch.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Go field names must be CamelCase. JSON field names must be camelCase. Other than capitalization of the initial letter, the two should almost always match. No underscores nor dashes in either.
+// This rule verifies the convention "Other than capitalization of the initial letter, the two should almost always match."
+// Examples (also in unit test):
+//     Go name      | JSON name    | match
+//     PodSpec        podSpec        true
+//     podSpec        podSpec        false
+//     PodSpec        spec           false
+//     Spec           podSpec        false
+//     JSONSpec       jsonSpec       true
+//     JSONSpec       jsonspec       false
+//     HTTPJSONSpec   httpJSONSpec   true
+// NOTE: this validator cannot tell two sequential all-capital words from one word, therefore the case below
```

Comment:
Does this also check that each field has been specified?  e.g. does empty match?

Reply:
Empty JSON name is considered special case that should be skipped instead of evaluated. Empty Go name with specified JSON name is considered not matching. I will add test cases for them.